components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -412,6 +427,7 @@ public Application updateApplication(ApplicationUpdateRequest updateRequest, Str
         oAuthConsumerAppDTO.setJwksURI(updateRequest.getJwksURI());
         Application application = buildResponse(oAuthConsumerAppDTO, tenantDomain);
         application.setSoftwareStatement(updateRequest.getSoftwareStatement());
+        application.setExtAllowedAudience(sp.getAssociatedRolesConfig().getAllowedAudience());
```

Comment:
can't we move this line to 417 into the if clause

Reply:
We cannot as we are creating the application object to build the response object in line number 428 only. In line number 417 we are at the point of updating the SP.

Also the allowed audience property in the response object will be null if we don't update the property in that specific request.